### PR TITLE
Implementing the Status service and data model

### PR DESCRIPTION
**Pull request by rtyler**
This pull request introduces the basics of the Status service and the necessary mechanisms to deliver some information to the `evergreen-client` for upgrading.

See also:

* [JENKINS-49809](https://issues.jenkins-ci.org/browse/JENKINS-49809)
* [JENKINS-49810](https://issues.jenkins-ci.org/browse/JENKINS-49810)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: '3'
+services:
+  db:
+    image: postgres:alpine
+    environment:
+      - 'POSTGRES_PASSWORD=grassisevergreener'
+      - 'POSTGRES_DB=evergreen_development'
+
+  backend:
+    image: jenkinsciinfra/evergreen
```
Comment (batmat) [inline docker-compose.yml]: Shouldn't it be `jenkins/evergreen`? https://github.com/jenkins-infra/evergreen/blob/master/Makefile#L1

Comment (batmat) [reply docker-compose.yml]: OK, my bad, I understand this is the services container. How about call it `jenkinsciinfra/evergreen-services` or something? (I'm still mostly on "I don't know what I'm doing" mode, so possibly I'm missing something).

File: services/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# http://editorconfig.org
+root = true
+
+[*]
+indent_style = space
```
Comment (batmat) [inline services/.editorconfig]: OK, I'm out. :D

**Comment by rtyler** [comment]
I need to do some work on making sure that the migrations wait for PostgreSQL to be fully online before attempting to execute, woops

**Comment by rtyler** [comment]
I think @batmat disappeared for the day, since he hasn't had objections to this pull request, I'm going to merge it with to avoid piling more commits onto this single pull request.

**Pull request merged**